src/testing/slim_check/gen.lean

```diff
@@ -0,0 +1,123 @@
+/-
+Copyright (c) 2020 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author(s): Simon Hudon
+-/
+
+import control.uliftable
+import system.random
+import system.random.basic
+
+/-!
+# Gen Monad
+
+This monad is used to formulate randomized computations with a parameter
+to specify the desired size of the result.
+
+This is a port of the Haskell QuickCheck library.
+
+## Main definitions
+  * `gen` monad
+
+## Local notation
+
+ * `i .. j` : `Icc i j`, the set of values between `i` and `j` inclusively;
+
+## Tags
+
+random testing
+
+## References
+
+  * https://hackage.haskell.org/package/QuickCheck
+
+-/
+
+universes u v
+
+/-- Monad to generate random examples to test properties with.
+It has a `nat` parameter so that the caller can decide on the
+size of the examples -/
+@[reducible]
+def gen (α : Type u) := reader_t (ulift ℕ) rand α
```

Comment:
`gen` is a pretty `gen`eric name to have in the root namespace. Is there a better name for this monad? Or maybe everything should be in the `slim_check` namespace?

Reply:
I see `sampleable` is in the `slim_check` namespace, so maybe that's the right solution here.